src/settings/PathConfigStorage.h

```diff
@@ -24,30 +24,48 @@ class PathConfigStorage
     const ExtruderTrain* support_infill_train;
     const ExtruderTrain* support_roof_train;
     const ExtruderTrain* support_bottom_train;
+
+    std::vector<GCodePathConfig> prime_tower_config_per_extruder; //!< Configuration for the prime tower per extruder.
+    std::vector<GCodePathConfig> prime_tower_config_per_extruder_layer0; //!< Configuration for the first layer of the prime tower per extruder.
+
+    GCodePathConfig support_infill_config; //!< The config used to print the normal support, rather than the support interface
+    GCodePathConfig support_infill_config_layer0; //!< The config used to print the first layer of the normal support
+    GCodePathConfig support_roof_config; //!< The config used to print the dense roofs of support.
+    GCodePathConfig support_roof_config_layer0; //!< The config used to print the dense roofs of support if they are on the first layer
 public:
     class MeshPathConfigs
     {
-    public:
+    private:
         GCodePathConfig inset0_config;
+        GCodePathConfig inset0_config_layer0;
         GCodePathConfig insetX_config;
+        GCodePathConfig insetX_config_layer0;
         GCodePathConfig skin_config;
-        GCodePathConfig ironing_config;
+        GCodePathConfig skin_config_layer0;
         GCodePathConfig perimeter_gap_config;
+        GCodePathConfig perimeter_gap_config_layer0;
+        GCodePathConfig infill_config_layer0; // combined infill layers can't be printed on the first layer
         std::vector<GCodePathConfig> infill_config;
 
+    public:
+        GCodePathConfig ironing_config;
```

Comment:
I'd like to keep these configs consistent. Right now half of them are private with their own `get***Config` functions (the ones that are different for the first layer), and the other half are public without such a function. This makes it inconsistent since the programmer has to remember which configs are public and which aren't.

So please also make the rest of them private and provide a `get***Config` function. Perhaps you can just give it no parameters since it doesn't vary by layer number.